src/ControllerStatusIO.c

```diff
@@ -315,6 +315,11 @@ BOOL Ros_Controller_IsAlarm()
         || (g_Ros_Controller.ioStatus[IO_ROBOTSTATUS_ALARM_USER]!=0) );
 }
 
+BOOL Ros_Controller_IsMajorAlarm()
+{
+    return (g_Ros_Controller.ioStatus[IO_ROBOTSTATUS_ALARM_MAJOR] != 0);
```

Comment:
I know it's not needed, but let's keep style consistent:

```suggestion
    return ((g_Ros_Controller.ioStatus[IO_ROBOTSTATUS_ALARM_MAJOR] != 0));
```